### PR TITLE
Update Summary page charts

### PR DESCRIPTION
**Pull request by maciekzygmunt**
Resolves L2B-8503 L2B-8588

### CODE REVIEW

**Review by torztomasz — CHANGES_REQUESTED** [review]

File: packages/frontend/src/server/features/scaling/tvl/utils/get-tvl-projects.ts [hunk 1]
```diff
@@ -133,21 +133,18 @@ function getCategory(
...
-  switch (category) {
+  switch (p.display.category) {
     case 'Validium':
     case 'Optimium':
       return 'ValidiumOrOptimiums'
     case 'Optimistic Rollup':
     case 'ZK Rollup':
       return 'Rollups'
+    case 'Other':
+      return 'Others'
```
Comment (torztomasz) [inline packages/frontend/src/server/features/scaling/tvl/utils/get-tvl-projects.ts:144]: make those expected as return type

**Review by torztomasz — CHANGES_REQUESTED** [review]

**Review by torztomasz — APPROVED** [review]

**Comment by lucadonnoh** [comment]
can you also make this chart use the same color for Validiums & Optimiums as in the summary page? for consistency

image

File: packages/frontend/src/components/chart/activity/activity-chart-hover.tsx [hunk 2]
```diff
@@ -93,3 +107,17 @@ export function ActivityChartHover(props: Props) {
...
 }
+
+function getColorByType(type?: ActivityChartType) {
+  switch (type) {
+    case 'Rollups':
+      return 'bg-indicator-rollups'
+    case 'ValidiumsAndOptimiums':
+      return 'bg-indicator-validiums-optimiums'
+    case 'Others':
+    case undefined:
```
Comment (torztomasz) [inline packages/frontend/src/components/chart/activity/activity-chart-hover.tsx:118]: why is this a case for undefined?

File: packages/frontend/src/components/chart/activity/scaling-summary-activity-chart.tsx [hunk 3]
```diff
@@ -56,14 +76,72 @@ export function ScalingSummaryActivityChart({ timeRange }: Props) {
...
+            {
+              name: 'Others',
+              color: 'bg-indicator-others',
+            },
+            {
+              name: 'Ethereum',
+              color: 'bg-indicator-ethereum',
+            },
+          ]}
+        />
```
Comment (torztomasz) [inline packages/frontend/src/components/chart/activity/scaling-summary-activity-chart.tsx:131]: we can keep it but composition would be nicer

File: packages/frontend/src/components/chart/core/chart-legend.tsx [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  return (
+    <div
+      className={cn(
+        'flex w-full flex-wrap justify-center gap-2',
+        props.className,
+      )}
+    >
+      {props.elements.map((e) => (
+        <div key={e.name} className="flex items-center gap-[3px]">
+          <div className={cn('size-[10px] rounded-sm', e.color)} />
```
Comment (torztomasz) [inline packages/frontend/src/components/chart/core/chart-legend.tsx:22]:
```suggestion
          <div className={cn('size-2.5 rounded-sm', e.color)} />
```

File: packages/frontend/src/components/chart/tvl/scaling-summary-tvl-chart.tsx [hunk 5]
```diff
@@ -50,6 +67,57 @@ export function ScalingSummaryTvlChart({
   )
 }
 
+function RecategorizedChart({
```
Comment (torztomasz) [inline packages/frontend/src/components/chart/tvl/scaling-summary-tvl-chart.tsx:70]: should be `RecategorisedTvlChart`

File: packages/frontend/src/server/features/scaling/activity/get-recategorized-activity-chart.ts [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+          validiumsAndOptimiumsEntry?.uopsCount ?? 0
+        const othersCount = othersEntry?.uopsCount ?? 0
+        const ethereumCount = ethereumEntry?.ethereumUopsCount ?? 0
+
+        return [
+          +timestamp,
+          rollupsCount,
+          validiumsAndOptimiumsCount,
+          othersCount,
+          ethereumCount,
```
Comment (torztomasz) [inline packages/frontend/src/server/features/scaling/activity/get-recategorized-activity-chart.ts:125]: what about tps?

File: packages/frontend/src/styles/globals.css [hunk 7]
```diff
@@ -36,6 +43,13 @@
     --icon-secondary: #5f6470;
     --overlay: #00000099;
     --divider: #3f4454;
+    --indicator-rollups: #db8bf7;
+    --indicator-validiums-optimiums: #3ac5e1;
+    --indicator-others: #dfd043;
+    --indicator-ethereum: #53a2ff;
+    --success-positive: #15ca60;
```
Comment (torztomasz) [inline packages/frontend/src/styles/globals.css:50]: not in this PR but change it to positive (accepted by Jan)

**Review by torztomasz — CHANGES_REQUESTED** [review]

**Review by torztomasz — APPROVED** [review]

**Pull request merged**